packages/@aws-cdk/aws-iot-actions-alpha/lib/open-search-action.ts

```diff
@@ -0,0 +1,59 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as iot from '@aws-cdk/aws-iot-alpha';
+import * as opensearch from 'aws-cdk-lib/aws-opensearchservice';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for Open Search.
+ */
+export interface OpenSearchActionProps extends CommonActionProps {
+  /**
+   * The unique identifier for the document you are storing.
+   */
+  readonly id: string;
+
+  /**
+   * The OpenSearch index where you want to store your data.
+   */
+  readonly index: string;
+
+  /**
+   * The type of document you are storing.
+   */
+  readonly type: string;
+}
+
+/**
+ * The action to write data to an Amazon OpenSearch Service domain.
+ */
+export class OpenSearchAction implements iot.IAction {
+  constructor(private readonly domain: opensearch.Domain, private readonly props: OpenSearchActionProps) {
```

Comment:
Probably my own lack of knowledge but why are the domain and props `private readonly`?

Reply:
that's the short format for defining the following
```
private const domain: opensearch.Domain;

constructor(domain: xx) {
  this.domain = domain;
}
```